File: backend/app/models/finding.py
from typing import Literal

from pydantic import BaseModel


class Finding(BaseModel):
    title: str

    severity: Literal[
        "CRITICAL",
        "HIGH",
        "MEDIUM",
        "LOW",
        "INFO",
        "WARNING",
        "ERROR",
        "UNKNOWN",
    ]

    category: str = "Other"

    file: str

    line: int

    message: str

    source: str

    snippet: str = ""

    explanation: str = ""

    recommendation: str = ""

    secure_code: str = ""

File: frontend/types/finding.ts
export type FindingSeverity =
  | "CRITICAL"
  | "HIGH"
  | "MEDIUM"
  | "LOW"
  | "INFO"
  | "WARNING"
  | "ERROR"
  | "UNKNOWN";

export interface Finding {
  title: string;
  severity: FindingSeverity;
  category: string;
  file: string;
  line: number;
  message: string;
  source: string;
  snippet: string;
  explanation: string;
  recommendation: string;
  secure_code: string;
}
